src/libraries/System.Runtime.InteropServices/ref/System.Runtime.InteropServices.cs

```diff
@@ -362,17 +362,17 @@ public sealed partial class ComObject : System.Runtime.InteropServices.IDynamicI
         System.Runtime.InteropServices.Marshalling.VirtualMethodTableInfo System.Runtime.InteropServices.Marshalling.IUnmanagedVirtualMethodTableProvider.GetVirtualMethodTableInfoForKey(System.Type type) { throw null; }
     }
     [System.Runtime.InteropServices.Marshalling.CustomMarshallerAttribute(typeof(System.Exception), System.Runtime.InteropServices.Marshalling.MarshalMode.UnmanagedToManagedOut, typeof(System.Runtime.InteropServices.Marshalling.ExceptionAsDefaultMarshaller<>))]
-    public static partial class ExceptionAsDefaultMarshaller<T> where T : struct
+    public static partial class ExceptionAsDefaultMarshaller<T> where T : unmanaged
```

Comment:
FWIW this would [not have produced a runtime exception](https://sharplab.io/#v2:EYLgtghglgdgNAFxAJwK7wCYgNQB8ACATAIwCwAUPgAwAE+xAdAEroJRgCmDAkjAh8gD2ABwDKAgG5QAxhwDODALIRkcgBYQANptgBzANwUKAemM0YgmhwAes4W0EwKAUVsd7URwEE5AEQ4AZhComgjKqhraAgA8igCeogho0ggAfAwAwo4SAggAKoIAqjCQMBC6HBgAFDAcAO40rnYOMFUAlG2G5CZmAkLINBmiABwAzADs4z00AYKCsQlJqCmp7V0U9ACcVQBE6oIhGDAA5Ag0yBwQ0mo0agIcO51GlAAsM3PReattNHV3FzQ8iAaOhSuVKjQAN4AXw2oxociWKRo8USyQQFEhFBoOLo8PotDkXWhQA===).

Reply:
@jkoritzinsky let me know that for these we want the constraints here.
> Any type that doesn't follow the `: unmanaged` constraints will have unexpected runtime behaviors